docker/Dockerfile.nvidia

```diff
@@ -114,74 +47,122 @@ ENV UCX_TLS=rc_x,sm,shm,cuda_copy,gdr_copy,cuda_ipc
+FROM sdk-base as nvc
+
+# Make devito env vars file and extras
 ADD docker/nvdashboard.json /app/nvdashboard.json
 
-RUN chmod +x /print-defaults /jupyter /tests /docker-entrypoint.sh && \
-    /venv/bin/jupyter serverextension enable dask_labextension 
-#    /venv/bin/jupyter lab workspaces import /app/nvdashboard.json
+# mpi4py
+RUN CFLAGS=-noswitcherror /venv/bin/pip install --no-cache-dir mpi4py && rm -rf ~/.cache/pip
+    # /venv/bin/jupyter lab workspaces import /app/nvdashboard.json
 
-## Environment Variables for OpenACC Builds
-# Reference: https://github.com/devitocodes/devito/wiki/FAQ#can-i-manually-modify-the-c-code-generated-by-devito-and-test-these-modifications
-# Set arch to PGI (pgcc)
-ENV DEVITO_ARCH="nvc" 
+ENV DEVITO_ARCH="nvc"
+ENV DEVITO_PLATFORM="nvidiaX"
 ENV DEVITO_LANGUAGE="openacc"
-ENV DEVITO_PLATFORM=nvidiaX
-# Options: [unset, 1] For PGI openacc; Should only be set after a first execution of the benchmark
-# ENV DEVITO_JIT_BACKDOOR=1 
-
-# Enable logging, Options: [unset, PERF, DEBUG]
-ENV DEVITO_LOGGING=DEBUG 
-#ENV PGI_ACC_TIME=1 
-
-## Create App user 
-# Set the home directory to our app user's home.
-ENV HOME=/app
-ENV APP_HOME=/app
-
-# Create the home directory for the new app user.
-# Create an app user so our program doesn't run as root.
-# Chown all the files to the app user.
-RUN mkdir -p /app && \
-    groupadd -r app && \
-    useradd -r -g app -d /app -s /sbin/nologin -c "Docker image user" app && \
-    chown -R app:app $APP_HOME && \
-    chown app:app /docker-entrypoint.sh && \
-    chown app:app /print-defaults && \
-    chown app:app /tests && \
-    chown app:app /jupyter
-    
-# Change to the app user.
-USER app
-WORKDIR /app/devito
+
+########################################################################
+# Build clang 14. This is followign the wiki:
+# https://en.wikibooks.org/wiki/LLVM_Compiler/Installation.
+# According to the wiki, clang should be built first with gcc then rebuilt with itself for openmp.
+########################################################################
+# Thid will only trigger if arch is clang since the final stage depends on it
+FROM sdk-base as clang
+
+## Install clang requirements
+RUN apt-get -y update && apt-get install -y -q libelf-dev libffi-dev cmake git gcc-multilib g++-multilib
+
+########################################################################
+# Build clang 14 with gcc
+########################################################################
+RUN git -c advice.detachedHead=false clone --depth 1 https://github.com/devitocodes/llvm-project.git
+
+RUN mkdir -p /llvm-project/build
+RUN cd /llvm-project/build && \
+    cmake ../llvm/ -DCMAKE_BUILD_TYPE=Release \
+    -DLLVM_TARGETS_TO_BUILD="X86;NVPTX" \
+    -DLLVM_ENABLE_PROJECTS="clang" \
+    -DLLVM_ENABLE_RUNTIMES="openmp" \
+    -DCLANG_OPENMP_NVPTX_DEFAULT_ARCH=sm_86 \
+    -DLIBOMPTARGET_NVPTX_COMPUTE_CAPABILITIES=35,37,50,52,60,61,70,75,86 \
```

Comment:
-DLIBOMPTARGET_NVPTX_COMPUTE_CAPABILITIES=all?

Reply:
Maybe better to have them explicitly unrolled....forget previous

Reply:
if "all" exists, I vote "all"